tests/src/test_esuuid.cpp

```diff
@@ -237,6 +237,27 @@ SCENARIO( "Mapping identifiable collections", "[uuid][comparison][collections]"
 			CHECK( collection.at(second) == -2 );
 		}
 	}
+	GIVEN( "a collection of strings as ID comparator, with UUIDs, identifying items" ) {
+		auto collection = std::map<std::string, EsUuid>{};
+		Identifiable first;
+		EsUuid firstId = EsUuid::FromString(first.UUID().ToString());
+		Identifiable second;
+		EsUuid secondId = EsUuid::FromString(second.UUID().ToString());
+		EsUuid otherId;
+		std::vector<Identifiable> items{ first, second };
+		std::string firstName = "one";
+		std::string secondName = "two";
+		collection.insert({ {firstName, firstId}, {secondName, secondId} });
```

Comment:
The tests fail because of this line, which is an instruction to create entries in a collection whose key is a string and whose value is a _copy_ of the provided `EsUuid`s.

Reply:
I tried pretty much everything to get that copy to have the same id, with clones and all but for some reason it woudn't work

Guess I'll just put the id itself there then

Reply:
It isn't going to matter how you construct the container when its value type is `EsUuid` (i.e. an actual object of that type and not a pointer to an object of that type), when the id is copied the copy is intentionally not going to compare equal with the source. You must follow up via `clone` to really say, "yes, I want these two IDs to have the same value."

If you'd like to track down all the copies your new code is doing, you can make the copy constructor noisy and have it write to std::out / files::logerror.

Reply:
So if I understood correctly the line 239 of PlayerInfo.cpp most be changed into a clone of the generated EsUuid::FromString() ?